stock_reception_screen_measuring_device/models/stock_reception_screen.py

```diff
@@ -30,13 +36,17 @@ class StockReceptionScreen(models.Model):
         store=True,
     )
 
-    @api.depends("product_packaging_id", "product_packaging_id.measuring_device_id")
+    @api.depends(
+        "current_move_product_id.packaging_ids.measuring_device_id",
+    )
     def _compute_scan_requested(self):
         for record in self:
-            record.scan_requested = (
-                record.product_packaging_id
-                and record.product_packaging_id.measuring_device_id
-            )
+            all_product_packagings = record.current_move_product_id.packaging_ids
+            record.scan_requested = False
+            for packaging in all_product_packagings:
+                if packaging.measuring_device_id:
+                    record.scan_requested = True
+                    break
```

Comment:
```suggestion
            record.scan_requested = any(packaging.measuring_device_id for packaging in all_product_packagings)
```

Reply:
Even if this is nicer to read, I think the for loop implementation is faster, because it does not evaluate the value every   record.